src/app/task/task-details/task-details.component.ts

```diff
@@ -90,8 +92,10 @@ export class TaskDetailsComponent implements OnInit, OnDestroy {
       const id = this.task?.id;
 
       if (id) {
-        this.getTask(id);
-        this.editable = !this.editable;
+        this.taskService.updateTask(id, editedTask).subscribe(() => {
```

Comment:
unsubscribe, also do you need to fetch the new task, if you already update it on the FE? when you refresh the page you should see the same thing you see after updating, no?

Reply:
When i refresh the page ngoninit gets run which calls the same getTask method, thats why the latest data gets shown on refresh. I call the getTask method in updateTask callback, so that the user doesnt have to manually refresh the page to see the latest data. Thats my understanding of it maybe im wrong. Toughts?

Reply:
But don't you have the latest data after modifying frontend? the form data updates reactively, so I don't see the need to fetch the task after updating, or is it not working like that?